velox/dwio/dwrf/common/Common.h

```diff
@@ -145,73 +147,69 @@ struct EncodingKeyHash {
+class DwrfStreamIdentifier : public StreamIdentifier {
  public:
-  static const StreamIdentifier& getInvalid() {
-    static const StreamIdentifier INVALID;
+  static const DwrfStreamIdentifier& getInvalid() {
+    static const DwrfStreamIdentifier INVALID;
     return INVALID;
   }
 
  public:
-  StreamIdentifier()
-      : EncodingKey(), column(MAX_UINT32), kind(StreamKind_DATA) {}
+  DwrfStreamIdentifier() : kind_(StreamKind_DATA) {}
 
-  /* implicit */ StreamIdentifier(const proto::Stream& stream)
-      : StreamIdentifier(
+  /* implicit */ DwrfStreamIdentifier(const proto::Stream& stream)
+      : DwrfStreamIdentifier(
             stream.node(),
             stream.has_sequence() ? stream.sequence() : 0,
             stream.has_column() ? stream.column() : MAX_UINT32,
             stream.kind()) {}
 
-  StreamIdentifier(
+  DwrfStreamIdentifier(
       uint32_t node,
       uint32_t sequence,
       uint32_t column,
-      proto::Stream_Kind pkind)
-      : StreamIdentifier{
-            node,
-            sequence,
-            column,
-            static_cast<StreamKind>(pkind)} {}
+      StreamKind kind)
+      : StreamIdentifier(velox::cache::TrackingId(node, kind).id()),
+        column_{column},
+        kind_(kind),
+        encodingKey_{node, sequence} {}
 
-  StreamIdentifier(
+  DwrfStreamIdentifier(
       uint32_t node,
       uint32_t sequence,
       uint32_t column,
-      StreamKind kind)
-      : EncodingKey(node, sequence), column{column}, kind(kind) {}
+      proto::Stream_Kind pkind)
+      : DwrfStreamIdentifier(
+            node,
+            sequence,
+            column,
+            static_cast<StreamKind>(pkind)) {}
 
-  ~StreamIdentifier() = default;
+  ~DwrfStreamIdentifier() = default;
 
-  bool operator==(const StreamIdentifier& other) const {
+  bool operator==(const DwrfStreamIdentifier& other) const {
     // column == other.column may be join the expression if all files
     // share the same new version that has column field filled
-    return node == other.node && sequence == other.sequence &&
-        kind == other.kind;
+    return encodingKey_ == other.encodingKey_ && kind_ == other.kind_;
   }
 
   std::size_t hash() const {
-    return std::hash<uint32_t>()(node) ^ std::hash<uint32_t>()(sequence) ^
-        std::hash<uint32_t>()(kind);
+    return encodingKey_.hash() ^ std::hash<uint32_t>()(kind_);
   }
 
-  uint32_t column;
-  StreamKind kind;
-
   std::string toString() const {
     return fmt::format(
-        "[node={}, sequence={}, column={}, kind={}]",
-        node,
-        sequence,
-        column,
-        static_cast<uint32_t>(kind));
+        "[id={}, node={}, sequence={}, column={}, kind={}]",
+        id_,
+        encodingKey_.node,
+        encodingKey_.sequence,
+        column_,
+        static_cast<uint32_t>(kind_));
   }
-};
 
-struct StreamIdentifierHash {
-  std::size_t operator()(const StreamIdentifier& si) const {
-    return si.hash();
-  }
+  uint32_t column_;
```

Comment:
shall we have getters instead of exposing members directly? they should ideally be immutable. Also public member should not use _